sdk/src/main/java/com/yellowpineapple/wakup/sdk/activities/OffersActivity.java

```diff
@@ -22,48 +21,55 @@
 
 public class OffersActivity extends OfferListActivity {
 
-    StaggeredGridView gridView;
     View navigationView;
     PullToRefreshLayout ptrLayout;
     View emptyView;
     boolean alreadyRegistered = false;
+    RecyclerView recyclerView;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-        ActionBar actionBar = getActionBar();
-        if (actionBar != null) {
-            // Retrieve the current set of display options
-            final int displayOptions = actionBar.getDisplayOptions();
-            // Determine which display options are enabled
-            final boolean isHomeAsUpEnabled = (displayOptions & ActionBar.DISPLAY_HOME_AS_UP) != 0;
-            if (isHomeAsUpEnabled && Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN_MR2) {
-                actionBar.setHomeAsUpIndicator(R.drawable.wk_actionbar_back_root);
-            }
-        }
 
         setContentView(R.layout.wk_activity_offers);
 
         injectViews();
     }
 
-    void injectViews() {
+    protected void injectViews() {
+        super.injectViews();
         emptyView = findViewById(R.id.emptyView);
-        gridView = ((StaggeredGridView) findViewById(R.id.grid_view));
+        recyclerView = (RecyclerView) findViewById(R.id.recycler_view);
         navigationView = findViewById(R.id.navigationView);
         ptrLayout = ((PullToRefreshLayout) findViewById(R.id.ptr_layout));
-        final View btnMap = findViewById(R.id.btnMap);
-        final View btnBigOffer = findViewById(R.id.btnBigOffer);
-        final View btnMyOffers = findViewById(R.id.btnMyOffers);
+        View bMap = null, bBigOffer = null, bMyOffers = null;
+        try {
+            bMap = findViewById(R.id.btnMap);
+        } catch(NoSuchFieldError e) {
+            Log.d("NOFIELD", "btnMap");
+        }
+        try {
+            bBigOffer = findViewById(R.id.btnBigOffer);
+        } catch(NoSuchFieldError e) {
+            Log.d("NOFIELD", "btnBigOffer");
+        }
+        try {
+            bMyOffers = findViewById(R.id.btnMyOffers);
+        } catch(NoSuchFieldError e) {
+            Log.d("NOFIELD", "btnMyOffers");
+        }
+        final View btnMap = bMap;
+        final View btnBigOffer = bBigOffer;
+        final View btnMyOffers = bMyOffers;
```

Comment:
This try/catch doesn't seems necessary. `findViewById` method will return `null` when a view can not be found by its id and not should not raise an exception.